pytest_inmanta/test_parameter/parameter.py

```diff
@@ -164,6 +165,15 @@ def pytest_addoption(cls, parser: "Parser") -> None:
             for param in parameters:
                 TestParameterRegistry.add_option(parser, group_name, param)
 
+class ValueSetBy(Enum):
+    """
+    This class is used to record how the value was provided for a test parameter.
+    """
+
+    DEFAULT_VALUE: str = "DEFAULT_VALUE"
+    CLI: str = "CLI"
+    ENV_VARIABLE: str = "ENV_VARIABLE"
+    UNSET: str = "UNSET"
```

Comment:
This doesn't seem to be used, is it? I would remove it, `Optional` (as you use it now) is perfect for that.

Reply:
Indeed, this is an artifact from a previous implementation that I forgot to remove